### PR TITLE
Expose symbol placement transition duration option

### PR DESCRIPTION
**Pull request by LukasPaczos**
Closes https://github.com/mapbox/mapbox-gl-native/issues/13402.

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/layers/TransitionOptions.java [hunk 1]
```diff
@@ -35,10 +48,27 @@ public TransitionOptions(long duration, long delay) {
    * @return a new transition property object
    */
   @Keep
+  @Deprecated
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/layers/TransitionOptions.java]: missing deprecated javadoc indicating alternative method to call

File: platform/android/src/native_map_view.cpp [hunk 2]
```diff
@@ -682,13 +682,15 @@ jni::Local<jni::Object<TransitionOptions>> NativeMapView::getTransitionOptions(J
     const auto transitionOptions = map->getStyle().getTransitionOptions();
     const auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(transitionOptions.duration.value_or(mbgl::Duration::zero())).count();
     const auto delay = std::chrono::duration_cast<std::chrono::milliseconds>(transitionOptions.delay.value_or(mbgl::Duration::zero())).count();
-    return TransitionOptions::fromTransitionOptions(env, duration, delay);
+    const auto enablePlacementTransitions = static_cast<const jboolean>(transitionOptions.enablePlacementTransitions);
```
Comment (tobrun) [inline platform/android/src/native_map_view.cpp]: `static_cast<const jboolean>(transitionOptions.enablePlacementTransitions);`can be simplified to 
 `(jboolean) transitionOptions.enablePlacementTransitions`

File: platform/android/src/style/conversion/transition_options.cpp [hunk 3]
```diff
@@ -7,7 +7,8 @@ namespace conversion {
 Result<jni::Local<jni::Object<TransitionOptions>>> Converter<jni::Local<jni::Object<TransitionOptions>>, mbgl::style::TransitionOptions>::operator()(jni::JNIEnv& env, const mbgl::style::TransitionOptions& value) const {
     return TransitionOptions::fromTransitionOptions(env,
         std::chrono::duration_cast<std::chrono::milliseconds>(value.duration.value_or(mbgl::Duration::zero())).count(),
-        std::chrono::duration_cast<std::chrono::milliseconds>(value.delay.value_or(mbgl::Duration::zero())).count()
+        std::chrono::duration_cast<std::chrono::milliseconds>(value.delay.value_or(mbgl::Duration::zero())).count(),
+        static_cast<jboolean>(value.enablePlacementTransitions)
```
Comment (tobrun) [inline platform/android/src/style/conversion/transition_options.cpp]: we can do a more simplified casting to jboolean

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/layers/TransitionOptions.java (hunk 1 above)
Comment (LukasPaczos) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/layers/TransitionOptions.java]: It's an internal API, but won't hurt to include an entry I guess :+1:

**Pull request merged**